enctests/testframework/generatetests.py

```diff
@@ -0,0 +1,144 @@
+# This is for generating tests from the markdown code.
+import os
+import argparse
+import yaml
+import re
+from pathlib import Path
+
+try:
+    from yaml import CSafeLoader as SafeLoader
+    from yaml import CSafeDumper as SafeDumper
+
+except ImportError:
+    from yaml import SafeLoader, SafeDumper
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '--root',
+        action='store',
+        default='..',
+        help='Where to look for *.md files to generate the config file.'
+    )
+
+    parser.add_argument(
+        '--output-config',
+        action='store',
+        dest='output_config_file',
+        default="test_configs/documentation_tests.yml",
+        help='Specify output file to write test wrappers to.'
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+def main():
+    args = parse_args()
+    rootdir = Path(args.root).absolute()
+
+    markdownfiles = []
+    for root, dirs, files in os.walk(rootdir):
+        for name in files:
+            if name.endswith(".md"):
+                markdownfiles.append(os.path.join(root, name))
+
+    allinfo = []
+
+    for markdownfile in markdownfiles:
```

Comment:
Would it make sense to create a function out of the markdown parsing? And one of the yaml creation below?